src/ty/variant.rs

```diff
@@ -222,4 +246,17 @@ where
     pub fn discriminant(&self) -> Option<u64> {
         self.discriminant
     }
+
+    /// Returns the documentation of the variant.
+    pub fn docs(&self) -> &[T::String] {
+        &self.docs
+    }
+
+    /// Clear the docs for this variant.
+    pub fn clear_docs(&mut self) {
+        self.docs.clear();
+        for field in &mut self.fields {
+            field.clear_docs()
+        }
```

Comment:
I am not entirely convinced if this strategy of first building up the entire documentation and then getting rid of it again represents an optimal API.

Reply:
The static alternative I considered was to provide a feature for the derive macro which would not capture the docs in the first place. However if you wanted metadata both with and without docs you would have to compile it twice, so I opted for the dynamic version since metadata serialization should usually be a one time thing.

Can you think of a better way?

Reply:
I've removed this clearing docs feature, for ease of getting this PR done. We can always add it back or an alternative as a follow up.

Reply:
https://github.com/paritytech/scale-info/issues/94